api/envoy/api/v2/cds.proto

```diff
@@ -657,6 +657,35 @@ message Cluster {
   // :ref:`lb_policy<envoy_api_field_Cluster.lb_policy>` field has the value
   // :ref:`LOAD_BALANCING_POLICY_CONFIG<envoy_api_enum_value_Cluster.LbPolicy.LOAD_BALANCING_POLICY_CONFIG>`.
   LoadBalancingPolicy load_balancing_policy = 41;
+
+  // [#not-implemented-hide:] If present, tells the client where to send load reports via LRS.
+  //
+  // If multiple clusters are configured to use the same LRS server, then the data for all of
+  // those clusters may be sent on the same LRS stream.
+  //
+  // Note that if LRS is configured this way, then there is no need for the LRS server to send
```

Comment:
From a federation perspective I think it would be better if the LRS server could still independently configure the interval, and potentially even turn off reports until they are needed. So I guess 2 questions:
1) Can we clarify potentially that duration can be a sentinel value (empty, 0, whatever) that means that no reports are sent?
2) And that this can either be configured in the CDS or LRS response?

WDYT?

Reply:
Playing devil's advocate for a moment, why is it better to configure the interval via LRS and via CDS for federation purposes?  Shouldn't federation be able to configure this via CDS?

This seems similar to the federation use-case we discussed for setting load balancing weights.  How did we wind up deciding to handle that use-case once we removed proto-merging from the UDPA federation design?  Could we apply whatever solution we used for that to this case as well?

If there's a good reason why this can be more easily done via LRS than via CDS, then maybe we should just leave the negotiation in the LRS stream where it is today and instead restrict this change to simply point the client at the right LRS server.  I'd prefer to avoid the complexity of trying to support two different ways to do the same thing.

Reply:
> then maybe we should just leave the negotiation in the LRS stream where it is today and instead restrict this change to simply point the client at the right LRS server. I'd prefer to avoid the complexity of trying to support two different ways to do the same thing.

+1. This reason I want this is because there are different implementations and a separation of concerns between CDS and LRDS, and IMO it's better for LRS to control the reporting rate, so my strong preference is to have that be done by LRS only as you propose.

Reply:
Thinking about this further... I think the reason I was trying to move this out of LRS in the first place is that as soon as we start configuring the LRS server in the CDS response, if we leave the negotiation in the LRS stream, then a misconfiguration could cause duplicate load reporting.  Consider the case where cluster A configures LRS server X and cluster B configures LRS server Y.  If the LRS streams to X and Y both request data for both clusters, then the data could wind up being double-reported.

Also, I think Harvey's proposal for UDPA was that use-cases like LRS were going to be handled by having the client publish the load-report resources on the single UDPA stream.  I'm not sure we've fully thought through how we were going to decide which management server was going to get which load-report data in a federation deployment.

We should probably talk this through in a bit more detail.  How about we have a short meeting about this once Harvey is back?

Reply:
SGTM